### ISSUE TITLE
Last File Received -> Last File Sent

### ISSUE THREAD

**Issue by fti7** [syncthing/syncthing#1883]
Would be nice to have an Information about which File was sent lately (Per Folder)
I know this is not really easy because of the P2P Feature.
But it could at least show the Filename from the last sent Blocks (Maybe with the Device Name which had requested this Blocks?)

BTW: 
Intially i thought this Field "Folder -> Last File Received" shows Recieved+Sent Files,
simply because im in German its says "Letzte Datei" (Last File)
I've updated it to "Letzte Datei empfangen" @ Transiflex

Best Regards,
Frank

**Comment by uok** [comment]
I think this can be closed as the translation is alread visible in 0.11.9

**Comment by AudriusButkevicius** [comment]
Not really, as this is asking for "Last file sent"

**Comment by uok** [comment]
yep, you are right

**Comment by fti7** [comment]
Yep, i think this Feature is very important for Users to see what Syncthing is exactly doing (Sent+Recieved & From/To Node(s))
I've put an $40 Bounty on this issue

**Comment by canton7** [comment]
In my understanding, there's currently no concept of "sending" a file. Other devices can request bits of a file from you (remember that you might not be the only device which has that file), but you don't know when that other device has finished downloading the file.

Also, where do you draw the line? Have you "Uploaded" a file if another device has finished downloading a file which you have, even if it didn't actually fetch and bits of it from you? Only if it actually fetched a bit from you? Only if it fetched the entire file from you? Etc...

**Comment by fti7** [comment]
Filename + a List with the devices that have requested at least one Block
I would recommend to visualize this in the good old Torrent P2P way to show which Block was sent from the current device (Then you see if it was the complete file or just a part) - of course it should only be shown if there is at least one block requested


![azureus-peers](https://cloud.githubusercontent.com/assets/7014664/8722281/170a08c4-2bc3-11e5-9e36-bb32951e2d87.png)

**Comment by AudriusButkevicius** [comment]
There are people with 10TB of data, so keeping the whole history throughout a process lifetime is unreasonable. 

Perhaps last 1000 requests...

Also requests not have a block index, they have an offset (which in the future will probably be replaced with just a hash) in which case where are no more files, you request a hash, we find it some random file and give it to you, in which case it becomes block count?
